pkg/workqueue/gcs/gcs.go

```diff
@@ -79,6 +80,14 @@ func (w *wq) Queue(ctx context.Context, key string, opts workqueue.Options) erro
 	}
 	writer.Metadata[notBeforeMetadataKey] = opts.NotBefore.UTC().Format(time.RFC3339)
 
+	// Extract trace context from the context and store it in metadata
+	if span := trace.SpanFromContext(ctx); span.SpanContext().IsValid() {
+		traceparent := span.SpanContext().TraceID().String() + "-" +
```

Comment:
nit: can you use a helper method for constructing the traceparent string, so we can add unit test for them, to see when a traceparent string would look like?